src/Aspire.Hosting/ApplicationModel/ResourceExtensions.cs

```diff
@@ -729,4 +729,36 @@ internal static string[] GetResolvedResourceNames(this IResource resource)
             return [resource.Name];
         }
     }
+
+    /// <summary>
+    /// Adds a deployment-specific image tag callback to a resource.
+    /// </summary>
+    /// <typeparam name="T">The resource type.</typeparam>
+    /// <param name="builder">The resource builder.</param>
+    /// <param name="callback">The synchronous callback that returns the deployment tag name.</param>
+    /// <returns>The resource builder.</returns>
+    [Experimental("ASPIRECOMPUTE001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
+    public static IResourceBuilder<T> WithDeploymentImageTag<T>(this IResourceBuilder<T> builder, Func<string> callback) where T : class, IResource
```

Comment:
Update the callback signature to take the context object.


```suggestion
    public static IResourceBuilder<T> WithDeploymentImageTag<T>(this IResourceBuilder<T> builder, Func<DeploymentImageTagCallbackAnnotationContext, string> callback) where T : class, IResource
```